### PR TITLE
Divyanshu | Test | Bank account is not getting link automatically when integrating bank from ledger

### PR DESCRIPTION
**Pull request by dvCodeWorld**
https://app.clickup.com/t/86cx6xt3w

## Summary by CodeRabbit

- **New Features**
	- Refined bank account connection and integration functionality
	- Enhanced state management for bank-related data

- **Bug Fixes**
	- Updated conditional rendering for bank account information display
	- Improved logic for handling bank list initialization

- **Refactor**
	- Simplified method signatures for bank-related components
	- Updated type definitions in bank transaction services

### CODE REVIEW

File: apps/web-giddh/src/app/ledger/ledger.component.ts [hunk 1]
```diff
@@ -1044,6 +1042,22 @@ export class LedgerComponent implements OnInit, OnDestroy {
                 }
             }
         });
+
+        this.settingIntegrationComponentStore.getAllBankAccountsList$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            this.isBankAccountConnected = false;
+            if (response?.body?.length) {
+                this.bankList = response?.body;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/ledger/ledger.component.ts]:
```suggestion
                this.bankList = response.body;
```

File: apps/web-giddh/src/app/ledger/ledger.component.ts [hunk 2]
```diff
@@ -1044,6 +1042,22 @@ export class LedgerComponent implements OnInit, OnDestroy {
...
+                if (response.body?.some(item => item.account?.uniqueName === (this.lc.accountUnq ?? this.selectedAccountUniquename))) {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/ledger/ledger.component.ts]:
```suggestion
                if (response.body.some(item => item?.account?.uniqueName === (this.lc.accountUnq ?? this.selectedAccountUniquename))) {
```

File: apps/web-giddh/src/app/ledger/ledger.component.ts [hunk 3]
```diff
@@ -1044,6 +1042,22 @@ export class LedgerComponent implements OnInit, OnDestroy {
...
+                    this.isBankAccountConnected = true;
+                } else {
+                    this.showBankLinkButton = response.body.some(bank => Object.keys(bank.account).length === 0);
+                    this.unlinkBankList = response.body.filter(bank => Object.keys(bank.account).length === 0);
+                }
+                if (this.referenceNumber !== null) {
+                    this.openBankLinkDialog()
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/ledger/ledger.component.ts]:
```suggestion
                    this.openBankLinkDialog();
```

File: apps/web-giddh/src/app/ledger/ledger.component.ts [hunk 4]
```diff
@@ -3223,7 +3237,7 @@ export class LedgerComponent implements OnInit, OnDestroy {
                 disableClose: true
             });
 
-            dialogRef.afterClosed().pipe(take(1), tap(response => { if (response) this.isBankAccountConnected = true; this.showBankLinkButton = false; this.getBankTransactions(); })).subscribe();
+            dialogRef.afterClosed().pipe(take(1), tap(response => { if (response) this.isBankAccountConnected = true; this.showBankLinkButton = false; this.getBankTransactions(); this.referenceNumber = null; })).subscribe();
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/ledger/ledger.component.ts]: format this code

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/ledger/ledger.component.ts (hunk 2 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/ledger/ledger.component.ts]: Done

File: apps/web-giddh/src/app/ledger/ledger.component.ts (hunk 4 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/ledger/ledger.component.ts]: Done

File: apps/web-giddh/src/app/ledger/ledger.component.ts (hunk 3 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/ledger/ledger.component.ts]: Done

**Review by dilpreetsinghofficial — APPROVED** [review]

**Pull request merged**